packages/clickable/components/clickable.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from "react";
+
+import { cx } from "emotion";
+import { outline } from "../style";
+
+export interface IClickableProps {
+  /**
+   * Children should be a HTML element.
+   */
+  children: React.ReactElement<HTMLElement>;
+  /**
+   * Action is a event handler for the onClick and onKeypress events
+   */
+  action: (event?: React.SyntheticEvent<HTMLElement>) => void;
+  /**
+   * The tabIndex is passed down and is the same as the native tabIndex
+   */
+  tabIndex?: number | string;
+}
+
+class Clickable extends React.PureComponent<IClickableProps, {}> {
+  public static defaultProps: Partial<IClickableProps> = {
+    tabIndex: -1
+  };
+
+  constructor(_) {
+    super(_);
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+  }
+
+  public render() {
+    const { children, action, tabIndex } = this.props;
+    const { className = "" } = children!.props;
```

Comment:
TIL about [`!.`](https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

Reply:
This is a Non-Null Assertation Operator

[read more here, should be the second section on the page](https://basarat.gitbooks.io/typescript/docs/options/strictNullChecks.html)

This is needed when something might be null, it is a typescript thing.

Reply:
But since I declared the children not to be optional anymore this is actually not needed anymore.